doc/man3/OSSL_PARAM_build.pod

```diff
@@ -0,0 +1,191 @@
+OSSL_PARAM structures.  The following B<TYPE> names are supported:
+
+=over 1
+
+=item *
+
+double
+
+=item *
+
+int
+
+=item *
+
+int32 (int32_t)
+
+=item *
+
+int64 (int64_t)
+
+=item *
+
+long int (long)
+
+=item *
+
+size_t
+
+=item *
+
+uint32 (uint32_t)
+
+=item *
+
+uint64 (uint64_t)
+
+=item *
+
+unsigned int (uint)
+
+=item *
+
+unsigned long int (ulong)
+
+=back
+
+OSSL_PARAM_BUILD_TYPE() are a series of macros designed to assist calling the
+OSSL_PARAM_build() function.
+In each case a parameter record is defined with the specified B<name> and
+B<value>.
+Each of these macros expands into a series of arguments appropriate for passing
+to the OSSL_PARAM_build() function which reduces the chance of parameters
+mistakes.
+Casts are also inserted to again reduce typing mistakes.
+
+OSSL_PARAM_BUILD_utf8_string(), OSSL_PARAM_BUILD_octet_string(),
+OSSL_PARAM_BUILD_BN() are macros that provide support
+for defining UTF8 strings, OCTET strings and big numbers.
+A parameter with B<name> is defined, storage of length B<len> is allocated and
+the B<value> copied.
+If the B<value> is NULL, the allocated memory will be zeroed.
+If B<len> is zero, the length will be calculated for
+OSSL_PARAM_BUILD_utf8_string() and OSSL_PARAM_BUILD_BN().
+For OSSL_PARAM_BUILD_octet_string() the length must be specified.
+
+OSSL_PARAM_build() allocates storage for and creates an array of OSSL_PARAM
+structures and their associated values.
+The parameter list must be terminated with a NULL.
+The order of the elements in the allocated OSSL_PARAM array are undefined.
+
+OSSL_PARAM_build_free() frees storage allocated for the B<params> array and
+the values the OSSL_PARAM structures reference.
+Only B<params> pointers returned from OSSL_PARAM_build() should be freed using
+this function.
+
+=head1 RETURN VALUES
+
+OSSL_PARAM_build() returns a populated OSSL_PARAM array, or NULL on failure.
+
+=head1 NOTES
+
+Using these functions introduces additional overhead when compared to raw
+OSSL_PARAM operations.
+
+=over 1
+
+=item *
+
+Memory for the array of OSSL_PARAM structures and the values referenced is
+dynamically allocated.
+
+=item *
+
+An extra copy is introduced for each value passed.
+
+=item *
+
+Individual OSSL_PARAM structures must be looked up by name.
+
+=back
+
+=head1 EXAMPLES
+
+This example creates an array of OSSL_PARAM containing two structures.
+One is a UTF8 string of length 32 which is named I<foo>.
+The other is an integer with the value 42.
+
+    #include <openssl/params.h>
+
+    OSSL_PARAM *params;
+    int bar_int;
+    char foo_buffer[1024];
+    char *foo_pointer = foo_buffer;
+    OSSL_PARAM *p;
+
+    /*
+     * Create the array containing the two parameter structures.
+     * It is automatically terminated with a third OSSL_PARAM_END structure.
+     */
+    params = OSSL_PARAM_build(
+        OSSL_PARAM_utf8_string("foo", NULL, 32),
+        OSSL_PARAM_int("bar", 42),
+        NULL
```

Comment:
This doesn't correspond to what's described in this manual.

Reply:
In what way?

Reply:
The comment at line 150 says ```OSSL_PARAM_END structure``` the text of the docs say NULL.

Reply:
It also says _automatically_ :)

I'll change.